src/main/java/io/cdap/plugin/salesforce/plugin/sink/batch/SalesforceRecordWriter.java

```diff
@@ -69,33 +80,65 @@ public SalesforceRecordWriter(TaskAttemptContext taskAttemptContext) throws IOEx
     AuthenticatorCredentials credentials = SalesforceConnectionUtil.getAuthenticatorCredentials(conf);
     bulkConnection = new BulkConnection(Authenticator.createConnectorConfig(credentials));
     jobInfo = bulkConnection.getJobStatus(jobId);
+    isFileUploadObject = FileUploadSobject.isFileUploadSobject(jobInfo.getObject());
+    if (isFileUploadObject) {
+      fileUploadObject = FileUploadSobject.valueOf(jobInfo.getObject());
+    }
   }
 
   @Override
-  public void write(NullWritable key, CSVRecord csvRecord) throws IOException {
+  public void write(NullWritable key, StructuredRecord record) throws IOException {
+    CSVRecord csvRecord = transformer.transform(record, fileUploadObject, ++count);
     csvBufferSizeCheck.reset();
     csvBufferSizeCheck.write(csvRecord);
 
-    if (csvBuffer.size() + csvBufferSizeCheck.size() > maxBytesPerBatch ||
+    if (csvBuffer.size() + csvBufferSizeCheck.size() + filesArraySize + getFileUploadSize(record) > maxBytesPerBatch ||
       csvBuffer.getRecordsCount() >= maxRecordsPerBatch) {
       submitCurrentBatch();
     }
 
+    if (isFileUploadObject) {
+      populateInputStreamMap(record);
+    }
     csvBuffer.write(csvRecord);
   }
 
+  private int getFileUploadSize(StructuredRecord record) {
```

Comment:
Better to have java doc stating the size limitation.

Reply:
done.

Reply:
done